### PR TITLE
[improvement](Nereids) add where Null rule to create empty relation as where false

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
explain shape plan select * from table2 where Null;
explain shape plan select * from table2 where false; 
in this case, null literal can be regard as same as false literal

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateFilter.java [hunk 1]
```diff
@@ -43,12 +44,13 @@ public class EliminateFilter implements RewriteRuleFactory {
     @Override
     public List<Rule> buildRules() {
         return ImmutableList.of(logicalFilter().when(
-                filter -> ExpressionUtils.containsType(filter.getConjuncts(), BooleanLiteral.class))
+                filter -> ExpressionUtils.containsType(filter.getConjuncts(), BooleanLiteral.class)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateFilter.java:47]: add some java ut for this rule

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateFilter.java:47]: done

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run External Regression

**Comment by LiBinfeng-01** [comment]
run External

**Comment by LiBinfeng-01** [comment]
./run external

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**